lua/sim/weapons/DefaultProjectileWeapon.lua

```diff
@@ -213,7 +212,7 @@ DefaultProjectileWeapon = ClassWeapon(Weapon) {
         projVelX = projVelX * multiplier
         projVelZ = projVelZ * multiplier
 
-        local targetPos, _
+        local targetPos
         local targetVelX, targetVelZ = 0, 0
```

Comment:
The `_` variable is needed, but the previous assignment was awkward, so this should make it clearer how it is used downstream
```suggestion
        local targetPos
        local targetVelX, _, targetVelZ = 0, nil, 0
```

Reply:
Why would the `_` be needed here if it isn't being assigned a value? I can appreciate amending L216 as you suggested for clarity, but that doesn't alter the change from `local targetPos, _` to `local targetPos`.

Happy to include the suggestion in the PR if you feel it's necessary.

Reply:
It's being assigned a value when we get the unit velocity in the function.
https://github.com/FAForever/fa/blob/8d4593bfaf57848fbb11c56f7d393248f4ab5ec2/lua/sim/weapons/DefaultProjectileWeapon.lua#L228
https://github.com/FAForever/fa/blob/8d4593bfaf57848fbb11c56f7d393248f4ab5ec2/lua/sim/weapons/DefaultProjectileWeapon.lua#L241
https://github.com/FAForever/fa/blob/8d4593bfaf57848fbb11c56f7d393248f4ab5ec2/lua/sim/weapons/DefaultProjectileWeapon.lua#L283

Reply:
Are we talking about the same lines?

https://github.com/FAForever/fa/pull/6864/files#diff-98546c6f8794a0bc3fcc3afbf785c38da3e5119f59251fb0ca86f4f17f4f5319L216-L217

Reply:
Are we talking about the same lines?

https://github.com/FAForever/fa/blob/8d4593bfaf57848fbb11c56f7d393248f4ab5ec2/lua/sim/weapons/DefaultProjectileWeapon.lua#L216

is changed in this PR to

https://github.com/FAForever/fa/blob/85d4a65093678373c0e04c287fb9e3586a586e7d/lua/sim/weapons/DefaultProjectileWeapon.lua#L215

but the line below of

https://github.com/FAForever/fa/blob/8d4593bfaf57848fbb11c56f7d393248f4ab5ec2/lua/sim/weapons/DefaultProjectileWeapon.lua#L217

remains unchanged as

https://github.com/FAForever/fa/blob/85d4a65093678373c0e04c287fb9e3586a586e7d/lua/sim/weapons/DefaultProjectileWeapon.lua#L216

Sanity check: you would like an extra `_` in the above two lines for better clarity of how other instances of `targetVelX` and `targetVelZ` are assigned?

As your suggestion:

```lua
local targetVelX, _, targetVelZ = 0, nil, 0
```

Reply:
It's not extra, we have `_` as a local variable so that when we get the velocity, the Y component isn't assigned to a global `_`.
I thought you removed it because it wasn't clear that it was being used for the (unused) Y component, so I suggested moving it alongside the X and Z initial assignments.

You could call it `targetVelY` and disable the "unused local" diagnostic to make it clear that it is assigned something, but in my opinion that reads slightly worse than `_`.

Reply:
Aaaahhhhh I understand now. Lua literally uses `_` as a variable and it's not some throwaway variable name. I'll use your suggestion :) Thanks for the clarity here!

Reply:
Ahhhh okay. That makes sense. You're right, we don't want to assign it to some global when it can be an otherwise unused local. Thanks for the clarity here.

Still getting the hang of Lua, I took `_` to mean a throwaway variable instead of literally `_` the `local`. Woops! :cake: